src/qibolab/instruments/icarusq.py

```diff
@@ -177,18 +11,23 @@ class MCAttenuator(Instrument):
     def connect(self):
         pass
 
-    def setup(self, attenuation: float, **kwargs):
-        """Assigns the attenuation level on the attenuator.
+    def play(self, *args):
+        pass
 
-        Arguments:
-            attenuation(float
-            ): Attenuation setting in dB. Ranges from 0 to 35.
-        """
-        import urllib3
+    @property
+    def attenuation(self):
+        http = urllib3.PoolManager()
+        res = http.request("GET", f"http://{self.address}/GETATT?")
+        return float(res._body)
 
+    @attenuation.setter
+    def attenuation(self, attenuation: float):
         http = urllib3.PoolManager()
         http.request("GET", f"http://{self.address}/SETATT={attenuation}")
 
+    def setup(self):
+        pass
```

Comment:
Indeed, `instrument.setup()` is used by 
https://github.com/qiboteam/qibolab/blob/974531413fdc53d9fcb476e375809ab5127bb029/src/qibolab/serialize.py#L101
to load the instrument settings from the runcard YAML. For this to work you have to explicitly call `load_instrument_settings` in the platform `create` method. Note that you don't have access to the physical instruments during platform creation, because `platform.connect()` is called after that. Therefore, following this approach requires caching the instrument settings in Python and uploading them only during connection.

If there are no settings relevant to the instrument, you can skip this step.